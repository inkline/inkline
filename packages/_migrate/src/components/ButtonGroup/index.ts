export { default as ButtonGroup } from '@inkline/inkline/components/ButtonGroup/ButtonGroup.vue';
export * from '@inkline/inkline/components/ButtonGroup/types';
