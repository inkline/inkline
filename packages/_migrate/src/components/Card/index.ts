export { default as Card } from '@inkline/inkline/components/Card/Card.vue';
