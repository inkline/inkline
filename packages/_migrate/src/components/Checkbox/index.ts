export { default as Checkbox } from '@inkline/inkline/components/Checkbox/Checkbox.vue';
