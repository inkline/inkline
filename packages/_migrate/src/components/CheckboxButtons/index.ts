export { default as CheckboxButtons } from '@inkline/inkline/components/CheckboxButtons/CheckboxButtons.vue';
export * from '@inkline/inkline/components/CheckboxButtons/types';
