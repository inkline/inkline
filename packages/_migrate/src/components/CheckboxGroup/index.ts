export { default as CheckboxGroup } from '@inkline/inkline/components/CheckboxGroup/CheckboxGroup.vue';
export * from '@inkline/inkline/components/CheckboxGroup/types';
