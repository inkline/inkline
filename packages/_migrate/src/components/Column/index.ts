export { default as Column } from '@inkline/inkline/components/Column/Column.vue';
