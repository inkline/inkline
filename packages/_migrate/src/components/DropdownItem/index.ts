export { default as DropdownItem } from '@inkline/inkline/components/DropdownItem/DropdownItem.vue';
