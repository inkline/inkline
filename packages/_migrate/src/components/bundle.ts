import type { Component } from 'vue';
import { Alert } from '@inkline/component-alert';
import { Badge } from '@inkline/component-badge';
import { Breadcrumb, BreadcrumbItem } from '@inkline/component-breadcrumb';
import { Button } from '@inkline/component-button';
import { Loader } from '@inkline/component-loader';

export const components: Record<string, Component> = {
    Alert,
    Badge,
    Breadcrumb,
    BreadcrumbItem,
    Button,
    Loader
};
