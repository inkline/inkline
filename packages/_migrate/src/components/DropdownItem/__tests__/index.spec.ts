import { fireEvent, render } from '@testing-library/vue';
import { DropdownItem } from '@inkline/inkline/components';

import { createTestingInklineOptionsProvide } from '@inkline/test-utils';
import { DropdownKey } from '@inkline/inkline';
import { ref } from 'vue';

describe('Components', () => {
    describe('DropdownItem', () => {
        const props = {};

        it('should be named correctly', () => {
            expect(DropdownItem.name).toEqual('DropdownItem');
        });

        it('should render correctly', () => {
            const wrapper = render(DropdownItem, {
                props,
                global: {
                    provide: {
                        ...createTestingInklineOptionsProvide()
                    }
                }
            });

            expect(wrapper.html()).toMatchSnapshot();
        });

        describe('computed', () => {
            describe('classes', () => {
                it('should add classes based on props', () => {
                    const wrapper = render(DropdownItem, {
                        props: {
                            active: true,
                            disabled: true,
                            plaintext: true
                        },
                        global: {
                            provide: {
                                ...createTestingInklineOptionsProvide()
                            }
                        }
                    });

                    expect(wrapper.container.firstChild).toHaveClass(
                        '-active',
                        '-disabled',
                        '-plaintext'
                    );
                });
            });

            describe('tabIndex', () => {
                it('should be -1 if disabled', () => {
                    const wrapper = render(DropdownItem, {
                        props: {
                            disabled: true,
                            ...props
                        },
                        global: {
                            provide: {
                                ...createTestingInklineOptionsProvide()
                            }
                        }
                    });

                    expect(wrapper.container.firstChild).toHaveAttribute('tabindex', '-1');
                });

                it('should be 1 otherwise', () => {
                    const wrapper = render(DropdownItem, {
                        props,
                        global: {
                            provide: {
                                ...createTestingInklineOptionsProvide()
                            }
                        }
                    });

                    expect(wrapper.container.firstChild).toHaveAttribute('tabindex', '0');
                });
            });
        });

        describe('methods', () => {
            describe('onClick()', () => {
                it('should call dropdown onItemClick', async () => {
                    const onItemClick = vi.fn();
                    const wrapper = render(DropdownItem, {
                        global: {
                            provide: {
                                ...createTestingInklineOptionsProvide(),
                                [DropdownKey as symbol]: {
                                    disabled: ref(false),
                                    color: ref('light'),
                                    size: ref('md'),
                                    onItemClick
                                }
                            }
                        },
                        props
                    });
                    const item = await wrapper.getByRole('menuitem');
                    await fireEvent.click(item);
                    expect(onItemClick).toHaveBeenCalled();
                });

                it('should not call dropdown onItemClick if not inside dropdown', async () => {
                    const onItemClick = vi.fn();
                    const wrapper = render(DropdownItem, {
                        global: {
                            provide: {
                                ...createTestingInklineOptionsProvide()
                            }
                        },
                        props
                    });
                    const item = await wrapper.getByRole('menuitem');
                    await fireEvent.click(item);
                    expect(onItemClick).not.toHaveBeenCalled();
                });
            });
        });
    });
});
