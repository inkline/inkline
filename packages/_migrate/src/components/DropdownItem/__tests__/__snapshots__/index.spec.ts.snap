// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Components > DropdownItem > should render correctly 1`] = `
"<div role="menuitem" tabindex="0" tag="div" class="dropdown-item">
  <!-- @slot default Slot for default dropdown item content -->
</div>"
`;
